src/metabase/api/card.clj

```diff
@@ -105,7 +106,8 @@
                                                    [:like :c.dataset_query (format "%%#%s%%" model-id)]]]]
                         :where [:and [:= :m.id model-id] [:not :c.archived]]})
        ;; now check if model-id really occurs as a card ID
-       (filter (fn [card] (some #{model-id} (-> card :dataset_query query/collect-card-ids))))))
+       (filter (fn [card] (some #{model-id} (-> card :dataset_query query/collect-card-ids))))
```

Comment:
Not your code, so don't need to action this.

nitpicks: 

- It bothers me that we're building the set each time inside the predicate (I think - maybe the Clojure compiler is smarter than I realize). Could hoist into a `model-id?` binding.
- I also think the threading here makes it harder to read - eyes need to start in the middle, scan right, then jump left again. One solution could be to go full-thread: 

```clojure
#(->> (:dataset_query %) query/collect-card-ids (some model-id?)))
```